java/core/src/main/java/com/microsoft/bond/SetBondType.java

```diff
@@ -148,6 +154,21 @@ protected final void serializeValue(SerializationContext context, Set<TElement>
     }
 
     @Override
+    protected final Set<TElement> deserializeValue(UntaggedDeserializationContext context) throws IOException {
+        final int count = context.reader.readContainerBegin();
+        Set<TElement> value = newDefaultValue(count);
```

Comment:
This doesn't map to capacity directly, as the latter is the size of hash table. Ideally, you should *compute* the initial capacity based on the expected size (which is the count) and load factor (default is 0.75).

Please see HashMap documentation (which is used to implement HashSet):

An instance of HashMap has two parameters that affect its performance: initial capacity and load factor. The capacity is the number of buckets in the hash table, and the initial capacity is simply the capacity at the time the hash table is created. The load factor is a measure of how full the hash table is allowed to get before its capacity is automatically increased. When the number of entries in the hash table exceeds the product of the load factor and the current capacity, the hash table is rehashed (that is, internal data structures are rebuilt) so that the hash table has approximately twice the number of buckets.

As a general rule, the default load factor (.75) offers a good tradeoff between time and space costs. Higher values decrease the space overhead but increase the lookup cost (reflected in most of the operations of the HashMap class, including get and put). The expected number of entries in the map and its load factor should be taken into account when setting its initial capacity, so as to minimize the number of rehash operations. If the initial capacity is greater than the maximum number of entries divided by the load factor, no rehash operations will ever occur.